Bee panics if `SIGINT` is sent during the "initializing NAT manager" phase

⚠️ Support requests in an issue-format will be closed immediately. For support, go to Swarm's [Discord](https://discord.gg/XGNBEKktkD).

### Context
Bee version: master branch commit: `3883de99`

### Summary
Bee panics if `SIGINT` is sent during the "initializing NAT manager" phase.

### Expected behavior
```
DEBU[2022-05-17T11:36:03-04:00] initializing NAT manager                     
^C
Process finished with the exit code 130 (interrupted by signal 2: SIGINT)
```

### Actual behavior
```
DEBU[2022-05-17T11:29:27-04:00] initializing NAT manager
^C
DEBU[2022-05-17T11:29:33-04:00] NAT manager initialized
DEBU[2022-05-17T11:29:33-04:00] kademlia: NewCollector(...) took 715.87µs
DEBU[2022-05-17T11:29:33-04:00] listener: batch factor 5
INFO[2022-05-17T11:29:33-04:00] waiting to sync postage contract data, this may take a while... more info available in Debug loglevel
ERRO[2022-05-17T11:29:33-04:00] got error interrupted, shutting down...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x19c pc=0x4c74a6f]

goroutine 1 [running]:
github.com/ethersphere/bee/pkg/node.(*Bee).Shutdown(0x0)
        /Users/nonameuser/Workspace/ethersphere/bee/pkg/node/node.go:954 +0x6f
github.com/ethersphere/bee/pkg/node.NewBee.func2()
        /Users/nonameuser/Workspace/ethersphere/bee/pkg/node/node.go:221 +0xba
github.com/ethersphere/bee/pkg/node.NewBee(0xc000dfe180, {0x4f76c79, 0x5}, 0xc000da6060, {0x53c08b8, 0xc00011a040}, 0xa, {0x53dda10, 0xc000ce0420}, 0xc001200060, ...)
        /Users/nonameuser/Workspace/ethersphere/bee/pkg/node/node.go:676 +0x4542
github.com/ethersphere/bee/cmd/bee/cmd.(*command).initStartCmd.func1(0xc000014840, {0xc0001b77a0, 0x2, 0x2})
        /Users/nonameuser/Workspace/ethersphere/bee/cmd/bee/cmd/start.go:158 +0x159e
github.com/spf13/cobra.(*Command).execute(0xc000014840, {0xc0001b7780, 0x2, 0x2})
        /Users/nonameuser/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc000014580)
        /Users/nonameuser/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/nonameuser/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/ethersphere/bee/cmd/bee/cmd.(*command).Execute(...)
        /Users/nonameuser/Workspace/ethersphere/bee/cmd/bee/cmd/cmd.go:159
github.com/ethersphere/bee/cmd/bee/cmd.Execute()
        /Users/nonameuser/Workspace/ethersphere/bee/cmd/bee/cmd/cmd.go:168 +0x45
main.main()
        /Users/nonameuser/Workspace/ethersphere/bee/cmd/bee/main.go:15 +0x1d

Process finished with the exit code 2
```

### Steps to reproduce
Sent `SIGINT` signal during the "initializing NAT manager" phase.

### Possible solution

--- a/pkg/node/node.go
+++ b/pkg/node/node.go
@@ -215,14 +215,14 @@ func NewBee(interrupt chan os.Signal, addr string, publicKey *ecdsa.PublicKey, s
 		tracerCloser:   tracerCloser,
 	}
 
-	defer func() {
+	defer func(b *Bee) {
 		if err != nil {
 			logger.Errorf("got error %v, shutting down...", err)
 			if err2 := b.Shutdown(); err2 != nil {
 				logger.Errorf("got error while shutting down: %v", err2)
 			}
 		}
-	}()
+	}(b)
 
 	stateStore, err := InitStateStore(logger, o.DataDir)
 	if err != nil {

